src/protocol.cpp

```diff
@@ -71,6 +73,8 @@ const static std::string allNetMessageTypes[] = {
     NetMsgType::CMPCTBLOCK,
     NetMsgType::GETBLOCKTXN,
     NetMsgType::BLOCKTXN,
+    NetMsgType::GETBFD,
+    NetMsgType::BFD,
```

Comment:
I would add a command VERBFD, where you send the hash of a BFD and then the node returns if the hash of that BFD matches their records. This makes it cheap to get consensus from another node. Alternatively, GETBFDH which just returns the hash (or make it an arg to GETBFD).